share/spice/skyscanner_flight_search/skyscanner_flight_search.js

```diff
@@ -0,0 +1,155 @@
+            var countryId = "SKY:" + placesById[current_route.DestinationId].SkyscannerCode; 
+            
+            if (response.index[countryId] >=0) {
+                image_array = response.results[response.index[countryId]].images;
+            } 
+            
+            if (Array.isArray(image_array) && image_array.length > 0) {
+                destination_city_image = image_array[0].url;
+            } else {
+                destination_city_image = "http://www.skyscanner.net/images/opengraph.png";
+            }
+
+            price = current_route.Price; 
+            price_age = current_route.QuoteDateTime;
+            // can use CityName or Name here:
+            destination_city = placesById[current_route.DestinationId].Name;
+            destination_airport = placesById[current_route.DestinationId].SkyscannerCode;
+            destination_code = destination_airport;
+            origin_code = placesById[current_route.OriginId].SkyscannerCode;
+            quote_ref = (current_route.QuoteIds) ? ((current_route.QuoteIds.length >1) ? current_route.QuoteIds[0]: current_route.QuoteIds) : '';
+            quote_date_time = (quote_ref != '') ? quotesById[quote_ref].QuoteDateTime : "N/A";
+
+            outbound_date = (quote_ref != '') ? quotesById[quote_ref].OutboundLeg.DepartureDate : "N/A";
+            return_date = (quote_ref != '') ? quotesById[quote_ref].InboundLeg.DepartureDate : "N/A";
+
+            return {
+                flight_price: price,
+                flight_price_age: price_age,
+                flight_destination_city: destination_city,
+                flight_destination_city_image: destination_city_image,
+                flight_destination_airport: destination_airport,
+                flight_destination_code: destination_code,
+                flight_origin_code: origin_code,
+                flight_quote: quote_ref,
+                flight_datetime: quote_date_time,
+                flight_outbound_date: outbound_date,
+                flight_return_date: return_date
+            };
+        }
+        var settings = {
+            "async": true,
+            "crossDomain": true,
+            // "url": "https://gateway.skyscanner.net/travel-api/v1/entities?external_ids=" + listOfCountryIds + "&enhancers=images&apikey=09fd8de5844d4b1d982a320ad5dee5b8",
+            "method": "GET",
+            "headers": {
+                "cache-control": "no-cache",
+                "postman-token": "cf3cb5f2-92ff-15a0-3331-6b014e83abba"
+            }
+        };
+
+        //$.ajax(settings).success(function (response) {
+          
+        $.getJSON('/js/spice/skyscanner_images/' + listOfCountryIds, function(response) {
+            var flights = listOfRoutes.map(function(route,index) { return build_flight_route(route, index, response); });
+            
+            // Fail if no flights to show
+            if (flights.length === 0) {
+                return Spice.failed('No flights found');
```

Comment:
The input here needs to be the Spice's `id`, not a message :)

e.g. `return Spice.failed('skyscanner_flight_search")`